### ISSUE TITLE
Framebuffer size is incorrect on Linux with HiDPI display

### ISSUE THREAD

**Issue by neauoire** [cloudhead/rx#52]
I've created a new issue so we could figure this one out on here. Could you remind me which line I should edit to better test the cursor positioning bug. I use a wacom intuos 3 tablet btw.

### Issue

When I move over the canvas with the wacom pencil, there is an offset from the screen position, and the canvas position.

![Screenshot from 2020-01-06 14-42-28@2x](https://user-images.githubusercontent.com/1455726/71843854-c8c60300-3092-11ea-98b1-0221d4cc774f.png)

**Comment by cloudhead** [comment]
Thanks for opening this! This bug is very likely due to the framebuffer being the incorrect size: the palette is also cut-off, and so is the session status bar. Effectively the framebuffer is exactly twice the resolution as the window, because `rx` is told that it should render 2 physical pixels for any given 1 "window" pixel.

If you make the following change:

```diff
- let hidpi_factor = win.hidpi_factor();
+ let hidpi_factor = 1.0;
```

You can force `rx` to create the right framebuffer size. If that works, at least we know the issue is dpi related, and we can go from there :+1:

**Comment by neauoire** [comment]
![Screenshot from 2020-01-06 16-55-51@2x](https://user-images.githubusercontent.com/1455726/71851843-72ae8b00-30a5-11ea-91d4-fb8c31ee5dab.png)

Yeah, that totally fixes it!

**Comment by cloudhead** [comment]
Amaaaaaaaazing. I think I will introduce a CLI flag for now to turn off hidpi detection until I figure out why this is happening. It looks like hidpi detection *is* working, since you have a retina screen and it's detecting `2`, but then the window being created is not big enough for the framebuffer :thinking: 

If the interface is too small, you should be able scale it with `:set scale = 2.0` - I'd be curious to know if that works correctly too with the current workaround.

**Comment by neauoire** [comment]
It does! And you know what, that's good enough for me :) We will be using Rx at Hundred Rabbits for our NES sprite assets, thank you.

**Comment by cloudhead** [comment]
Awesome. Excited to see what you create, and feel free to provide feedback & ideas on the issues list!

I'll leave this issue open until I fix this properly.

:boat:

**Comment by neauoire** [comment]
I'll definitely be asking for `.chr` export, but we'll get to that later. 
This will be just fine for now, keep up the good work, Rx is a beauty.

**Comment by cloudhead** [comment]
This should be fixed now with #54 merged. If you could confirm, would be great. (the `--features compatibility` flag is no longer necessary)

**Comment by cloudhead** [comment]
Fixed.
